### ISSUE TITLE
Reindexing wipes out other settings

### ISSUE THREAD

**Issue by mythsam** [algolia/algoliasearch-laravel#73]
When running reindex(), it overwrites the settings which is not being specified in the class $algoliaSettings variable. I believe we should preserve other settings if they exist in Algolia and are not specified on $algoliaSettings. 

The area to fix should be in $modelHelper->getIndicesTmp(), when we initiating the tmp index, we should clone the original index's setting over.

**Comment by maxiloc** [comment]
Yes that would be doable. Maybe an option is the best way to go.

**Comment by Pautomagi** [comment]
Any updates on this?

**Comment by maxiloc** [comment]
I'll try to work on it next week.
Feel free to submit a PR before then.

**Comment by Pautomagi** [comment]
Thanks!

I did a tempoary thing by calling setSettings in the Trait for now.

Something like this.
```                
if ($settings == true) {
    $index->setSettings($modelHelper->getSettings($this));
}
```

**Comment by maxiloc** [comment]
Fixed with 90ba259d1831648a3bff5a37b293a2fa7eea7e81
